File: Virtual Machine/Nova.VirtualMachine/Nova.VirtualMachine/ByteBlock.cpp
#include "ByteBlock.h"

void ByteBlock::Deserialize(BinaryReader& reader)
{
	int size = reader.Read<int>();

	for (int i = 0; i < size; i++)
	{
		instructions.push_back(reader.Read<int>());
	}

	this->localesCount = reader.Read<int>();
}

File: Compiler/Nova.VirtualMachine/Members/ByteBlock.cs
using Nova.Utils.IO;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Nova.VirtualMachine.Members
{
    public class ByteBlock
    {
        public int LocalesCount
        {
            get;
            set;
        }
        public List<int> Instructions
        {
            get;
            set;
        }
        public ByteBlock()
        {
            this.Instructions = new List<int>();
        }
        public void Deserialize(CppBinaryReader reader)
        {
            int size = reader.ReadInt32();

            for (int i = 0; i < size; i++)
            {
                this.Instructions.Add(reader.ReadInt32());
            }
            this.LocalesCount = reader.ReadInt32();
        }
    }
}
